mutiny-core/src/logging.rs

```diff
@@ -1,9 +1,90 @@
+use std::sync::{
+    atomic::{AtomicBool, Ordering},
+    Arc, Mutex,
+};
+
+use crate::{error::MutinyError, utils::sleep};
 use chrono::Utc;
+use gloo_utils::format::JsValueSerdeExt;
 use lightning::util::logger::{Level, Logger, Record};
 use log::*;
+use rexie::{ObjectStore, Rexie, TransactionMode};
+use wasm_bindgen::JsValue;
+use wasm_bindgen_futures::spawn_local;
+
+pub(crate) const LOGGING_DATABASE_NAME: &str = "logging";
+pub(crate) const LOGGING_OBJECT_STORE_NAME: &str = "log_store";
+pub(crate) const LOGGING_KEY: &str = "logs";
+
+#[derive(Default, Clone)]
+pub struct MutinyLogger {
+    should_write: bool,
```

Comment:
`should_write` is a little ambiguous, something like `should_write_to_storage` would be more clear